src/Container.php

```diff
@@ -397,7 +397,7 @@ protected function getRaw($key)
 
 		$aliasKey = $this->resolveAlias($key);
 
-		if ($aliasKey != $key && isset($this->dataStore[$aliasKey]))
+		if ($aliasKey !== $key && isset($this->dataStore[$aliasKey]))
```

Comment:
This one we might need to hold off for 2.0.  If I'm reading correctly, this will force case sensitivity for keys (FWIW Symfony is enforcing case sensitive keys in Symfony 4 and started throwing deprecation notices for non-case matching scenarios in a recent release).